Feat/profile page

frooooooo7 opened the pull request.

frooooooo7 committed:
feat: created SideMenu's component prototype (just design)

frooooooo7 committed:
feat: user's profile (displaying and editing user's info)

frooooooo7 committed:
feat: add ConfirmPopup component to confirm user's info changes.

frooooooo7 committed:
refactor: some changes in api requests.

frooooooo7 committed:
feat: create PaymentTab form for user's payment informations.

frooooooo7 committed:
refactor: profile page server-side rendering

frooooooo7 committed:
feat: add validation to User's form.

refactor:
- change state management
- create FormField component reduce code duplication.

Finnick223 commented on .vscode/settings.json:
```diff
@@ -8,5 +8,8 @@
-  ]
+  ],
+  "[typescriptreact]": {
+    "editor.defaultFormatter": "vscode.typescript-language-features"
```
why change formatter from prettier??

Finnick223 commented on middleware.ts:
instead of making middleware for profile page it would be better to make profile page as sub route of dashboard and it would automatically be for logged in users

Finnick223 commented on enums/routes.enums.ts:
```diff
@@ -4,6 +4,7 @@ export enum UnAuthorizedRoute {
 }
 export enum AuthorizedRoute {
   Dashboard = '/dashboard',
+  Profile = '/user',
```
Profile = '/dashboard/user',

Finnick223 commented on app/user/tabs/DataTab.tsx:
```diff
@@ -0,0 +1,126 @@
+                isOpen={isModalOpen}
+                onConfirm={handleConfirm}
+                onCancel={() => setIsModalOpen(false)} />
+            <ToastContainer />
```
i think ToastContainer should be higher in hierarchy (maybe route layout or dashboard layout?)

Finnick223 commented on app/user/page.tsx:
user page should be part of dashboard so route dashboard/user/page

Finnick223 commented on app/user/layout.tsx:
```diff
@@ -0,0 +1,18 @@
+}>) {
+  return (
+    <div className='flex h-screen '>
+      <SideMenu />
```
move SideMenu to dashboard layout because it will be used across all dashboard pages

Finnick223 commented on app/user/components/FormField.tsx:
```diff
@@ -0,0 +1,30 @@
+    id: string;
+    label: string;
+    type?: string;
+    register: any;
```
maybe we can get rid of this any type
look at 
https://stackoverflow.com/questions/76803987/what-type-to-use-for-register-from-the-react-hook-form-register-typescript-type/77636341#77636341

frooooooo7 committed:
refactor:
- move user page to be part of dashboard
- move ToastContainer and SideMenu to layout.
- fix type of registration in FormField
- move logout button to SideMenu

Finnick223 commented on app/dashboard/user/[id]/components/FormField.tsx:
```diff
@@ -0,0 +1,41 @@
+interface FormFieldProps<T extends FieldValues> {
+  id: Path<T>;
+  label: string;
+  type?: string;
```
try HTMLInputTypeAttribute to make it more type safe

Finnick223 commented on app/dashboard/user/[id]/components/FormField.tsx:
```diff
@@ -0,0 +1,41 @@
+}: FormFieldProps<T>) => {
+  return (
+    <div>
+      <Label className='text-xl'>{label}</Label>
```
Label should have included htmlFor={id as string} prop

Finnick223 commented on app/dashboard/user/[id]/tabs/DataTab.tsx:
```diff
@@ -0,0 +1,140 @@
+    try {
+      await axiosInstance.put(`${ApiRoutes.User(userID)}`, formData);
+      setIsModalOpen(false);
+      window.location.reload();
```
It's causing toast do disappear
try reloading/redirecting based on nextjs routing if it wont work we're gonna use state management based on react-query

Finnick223 commented on app/dashboard/user/[id]/tabs/PaymentTab.tsx:
```diff
@@ -0,0 +1,116 @@
+        formData,
+      );
+      setIsModalOpen(false);
+      window.location.reload();
```
Same as in DataTab

frooooooo7 committed:
fix:
- change type to HTMLInputTypeAttribute in FormField.
- add htmlFor for label in FormField.
- fix toast disappearing

Finnick223 reviewed (changes requested):
Your state management from scratch lgtm. 
In future we're gonna add TanStack Query but for now add responsiveness and task will be done:)

frooooooo7 committed:
feat: responsiveness for SideMenu and ProfilPage.

Finnick223 commented on app/dashboard/layout.tsx:
```diff
@@ -1,18 +1,19 @@
 import { ToastContainer } from 'react-toastify';
 
-export default function DashboardLayout({
+export default function ProfilePageLayout({
```
it's DashboardLayout

frooooooo7 committed:
feat:
- improve navigation
- highlight active link in SideMenu

Finnick223 reviewed (changes requested):
UI looks good, change layout function name and it's ready to merge

frooooooo7 committed:
refactor: rename ProfilPageLayout to DashboardLayout.

Finnick223 committed:
Merge branch 'develop' into feat/profile_page and resolve conflicts

- merge develop into feat/profile_page
- make necessary changes to get rid of conflicts and integrate code
- improve structure, modularity and naming
- fix security issue related to visible id in url on profile page

Finnick223 committed:
fix: react-toastify works on ssr
